source/element_parameters.md

```diff
@@ -0,0 +1,225 @@
+
+
+## Naming and Inheriting Parameter Groups
+
+Any group can be given a **name** and the values can be used in another group of the same type
+using **import**.
+For example:
+```{code} yaml
+Aperture:
+  name: ap1
+  x_limit: [-0.03, 0.03]
+```
+The above defines an aperture with the name **ap1**. 
+```{code} yaml
+Aperture:
+  name: ap2
+  inherit: ap1
+  y_limit: [-0.02, 0.02]
+```
+And the above defines a new aperture group which inherits from **ap1**.
+
+Naming a parameter group is only needed if the parameter group is defined outside of an element.
+```{code} yaml
+element:
+  name: q1
+  Aperture: 
+    x_limit: [-0.03, 0.03]
+    y_limit: [-0.02, 0.02]
+```
+And an element can inherit a parameter group from another element:
+```{code} yaml
+element:
+  name: q2
+  Aperture:
+    inherit: q1.Aperture
+```
+
+For an element to inherit all parameter groups from another element, just inherit the element itself:
+```{code} yaml
+element:
+  name: q3
+  inherit: q2
+```
+
+## Aperture Parameter Group
+
+Components:
+```{code} yaml
+x_limit
+y_limit
+aperture_shape
+aperture_at
+aperture_shifts_with_body
+wall2d
+material
+thickness
+```
+
+### aperture_at
+
+The aperture location is set by the `aperture_at` parameter. Possible values are
+```{code} yaml
+ENTRANCE_END   # Body entrance end (default)
+CENTER         # Element center
+EXIT_END       # Body exit end
+BOTH_ENDS      # Both ends
+NOWHERE        # No location
+EVERYWHERE     # Everywhere
+```
+The default is `ENTRANCE_END`.
+
+### aperture_shape
+
+```{figure} figures/apertures.svg
+:width: 90%
+:name: f:aperture
+
+A) RECTANGULAR and ELLIPTICAL apertures. As drawn, `x_limit[1]` and `y_limit[1]` are 
+negative and `x_limit[2]` and `y_limit[2]` are positive. B) The `wall2d` aperture is defined
+by a set of vertices.
+```
+
+The `aperture_shape` parameter selects the shape of the aperture. Possible values are:
+```{code} yaml
+RECTANGULAR   # Rectangular shape.
+ELLIPTICAL    # Elliptical shape.
+WALL2D        # Shape defined by set of vertices.
+CUSTOM_SHAPE  # Shape defined outside of the lattice standard.
+```
+
+### x_limit and y_limit
+
+For `RECTANGULAR` and `ELLIPTICAL` shapes the `x_limit` and `y_limit` parameters are
+used to calculate the aperture as shown in {numref}`f:aperture`A. 
+For an `ELLIPTICAL` aperture, a particle with position {math}`(x, y)` is outside of the aperture if any 
+one of the following four conditions is true:
+```{code}
+  1) x < 0 and y < 0 and (x/x_limit[1])^2 + (y/y_limit[1])^2 > 1 
+  2) x < 0 and y > 0 and (x/x_limit[1])^2 + (y/y_limit[2])^2 > 1
+  3) x > 0 and y < 0 and (x/x_limit[2])^2 + (y/y_limit[1])^2 > 1
+  4) x > 0 and y > 0 and (x/x_limit[2])^2 + (y/y_limit[2])^2 > 1
+```
+For a `RECTANGULAR` aperture the corresponding four conditions are:
+```{code}
+  1) x < x_limit[1]
+  2) x > x_limit[2]
+  3) y < y_limit[1]
+  4) y > y_limit[2]
+```
+
+Default values for the limits are `[-Inf, Inf]` for both `x_limit` and `y_limit`.
```

Comment:
Recommend adding an explicit specification of units here.  (Presumably meters.)

Reply:
See the new introduction for a discussion on units. One thing that should be discussed is the units for angles and phases.

Reply:
Done.